server/src/service/service.rs

```diff
@@ -0,0 +1,135 @@
+//  org.freedesOnceCellktop.Secret.Service
+
+use std::collections::HashMap;
+
+use oo7::portal::{Item, Keyring, Secret};
+use serde::{Deserialize, Serialize};
+use zbus::{dbus_interface, zvariant, Connection, Error, ObjectServer, SignalContext};
+use zvariant::{ObjectPath, OwnedObjectPath, OwnedValue, Type, Value};
+
+use crate::service::collection::Collection;
+use crate::service::session::Session;
+use crate::KEYRING;
+
+#[derive(Deserialize, Serialize, Type, Debug)]
+pub struct Service {
+    collections: Vec<Collection>,
+}
+
+#[dbus_interface(name = "org.freedesktop.Secret.Service")]
+impl Service {
+    pub async fn open_session(
+        &self,
+        algorithm: &str,
+        input: Value<'_>,
+    ) -> zbus::fdo::Result<(OwnedValue, OwnedObjectPath)> {
+        let session = Session::new(ObjectPath::try_from("/s1").unwrap().into()).await;
+        Ok((
+            OwnedValue::try_from(input).unwrap(),
```

Comment:
Errors that can happen due to invalid user input should not use `.unwrap()`. I'd recommend using a `Result<T, E>` type here to make error handling easier. While working on improving the code, I'd recommend turning on `clippy::unwrap_used` and `clippy::expect_used` to find other places where this may be an issue.

Reply:
Got it, thanks!